### PR TITLE
feat: Add random timezone to aggregate fuzz test

### LINKED ISSUE #13302: Add randomized timezones for timestamps
### Is your feature request related to a problem or challenge?

It is mentioned in #13280 to add randomized timezones.

### Describe the solution you'd like

Allow the timezones to be randomized in the fuzz testing.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonathanc-n**
## Which issue does this PR close?

Closes #13302 .

## What changes are included in this PR?
Added random timezones to fuzz test for timestamps

### CODE REVIEW

**Review by LeslieKid — COMMENTED** [review]
👍This PR is great.

The only thing i am concerned is -- the `timestamp` part in `baseline_config()` function in [aggr_fuzz.rs](https://github.com/apache/datafusion/blob/main/datafusion/core/tests/fuzz_cases/aggregate_fuzz.rs#L175C9-L187C11) is:
```rust
ColumnDescr::new("timestamp_s", DataType::Timestamp(TimeUnit::Second, None)),
ColumnDescr::new(
    "timestamp_ms",
    DataType::Timestamp(TimeUnit::Millisecond, None),
),
ColumnDescr::new(
    "timestamp_us",
    DataType::Timestamp(TimeUnit::Microsecond, None),
),
ColumnDescr::new(
    "timestamp_ns",
    DataType::Timestamp(TimeUnit::Nanosecond, None),
),
```
The implementation will generate the timezone randomly. So I think the `None` here may confuse the reader. Maybe we can add some comments here, or use `generate_timezone()` function to replace the `None` directly to make the code clear?

**Comment by jonathanc-n** [comment]
@LeslieKid Yeah I can definitely add a comment, as using generate_timezone() directly in the function makes it difficult to pass into the `generate_primitive_array`, as that function is used for other primitive types.

File: datafusion/core/tests/fuzz_cases/aggregate_fuzz.rs [hunk 1]
```diff
@@ -172,6 +172,8 @@ fn baseline_config() -> DatasetGeneratorConfig {
         ColumnDescr::new("time32_ms", DataType::Time32(TimeUnit::Millisecond)),
         ColumnDescr::new("time64_us", DataType::Time64(TimeUnit::Microsecond)),
         ColumnDescr::new("time64_ns", DataType::Time64(TimeUnit::Nanosecond)),
+        // `None` is passed in here however when generating the array, it will generate
```
Comment (alamb) [inline datafusion/core/tests/fuzz_cases/aggregate_fuzz.rs:175]: 💯

File: test-utils/src/array_gen/primitive.rs [hunk 2]
```diff
@@ -17,8 +17,9 @@
 
 use arrow::array::{ArrayRef, ArrowPrimitiveType, PrimitiveArray, UInt32Array};
 use arrow::datatypes::DataType;
-use rand::rngs::StdRng;
-use rand::Rng;
+use chrono_tz::{Tz, TZ_VARIANTS};
```
Comment (alamb) [inline test-utils/src/array_gen/primitive.rs:20]: TIL: https://docs.rs/chrono-tz/latest/chrono_tz/static.TZ_VARIANTS.html 👍

**Review by alamb — APPROVED** [review]
Looks great to me -- thank you @jonathanc-n  and @LeslieKid

**Pull request merged**